### PR TITLE
adding a new gnssro converter: NETCDF profiles to IODA

### PR DESCRIPTION
**Pull request by hailingz**
## Description

This PR adds a new GNSSRO converter.
It reads GNSSRO data from the AWS open data registry.
The AWS open data registry contains post-processed GNSSRO data from multiple processing centers (UCAR/COSMIC, ROMSAF, JPL). This converter focuses on the conversion of "atmPrf" in NETCDF. With this, we can use the post-processed RO data while other data feed (e.g., WMO bufr, NCEP bufr) are all near real-time data.

data available at: https://registry.opendata.aws/gnss-ro-opendata/
documents available at: https://github.com/gnss-ro/aws-opendata

NOTES: 

1. The original atmprf has more than 3000 levels vertically. The profiles are thinned in this converter using a changing vertical interval. 
Specifically, the interval between layers increasing from  ~120m near surface to ~310m at 40km. The interval is fixed to 310m above 40km.
Thanks @BenjaminRuston  for your help on the initial codes.
Thanks @haydenlj  for your help in the thinning. 

2. This converter also works for the UCAR/CDAAC atmprf with minor changes (so,e variable names may be different).

## What problem does it fix? What new capability does it add?
adds a new GNSSRO converter.

## Acceptance Criteria (Definition of Done)
pass ctests:
ctest -R gnssro_AWSopendataNetcdf_conv

## Dependencies
N/A

## Impact
N/A

### CODE REVIEW

**Comment by BenjaminRuston** [comment]
would like to get this in quickly but should try to minimize the redundancy

see if I can maybe help you reuse the code between these

**Comment by hailingz** [comment]
> redundancy

I use  "lib_python.def_jedi_utils" for the function  "concat_obs_dict" which was duplicated in a few of python converters. 
Not sure what other "redundancy" related to this PR.

**Comment by fcvdb** [comment]
@haydenlj Could you review this PR, when you have some time? Thanks.

File: src/gnssro/gnssro_AWSopendataNetcdf2ioda.py [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+    print("Reading: %s" % input_file)
+
+    f = h5py.File(input_file, 'r')
+    obs_data = get_opendata(f, add_qc, record_number=record_number)
+    f.close()
+    return obs_data
+
+
+# assign satellite identifier
+def get_WMO_satellite_ID(filename):
```
Comment (BenjaminRuston) [inline src/gnssro/gnssro_AWSopendataNetcdf2ioda.py:184]: this is also an attribute the file:  `f.attrs['leo']`

but we should request from AER to add the WMO satellite ID it is pretty standard

File: src/gnssro/gnssro_AWSopendataNetcdf2ioda.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    day = f.attrs['day'][0]
+    hour = f.attrs['hour'][0]
+    minute = f.attrs['minute'][0]
+    second = f.attrs['second'][0]
+
+    dtg = ("%4i-%.2i-%.2iT%.2i:%.2i:%.2iZ" % (year, month, day, hour, minute, second))
+    this_datetime = datetime.strptime(dtg, "%Y-%m-%dT%H:%M:%SZ")
+    time_offset = round((this_datetime - epoch).total_seconds())
+    profile_meta_data['dateTime'] = np.int64(time_offset)
+    WMO_sat_ID = get_WMO_satellite_ID(f.filename)
```
Comment (BenjaminRuston) [inline src/gnssro/gnssro_AWSopendataNetcdf2ioda.py:395]: here can pass the `f.attrs['leo']`

```suggestion
    WMO_sat_ID = get_WMO_satellite_ID(f.attrs['leo'])
```

Comment (hailingz) [reply src/gnssro/gnssro_AWSopendataNetcdf2ioda.py]: done!

File: src/gnssro/gnssro_AWSopendataNetcdf2ioda.py (hunk 1 above)
Comment (hailingz) [reply src/gnssro/gnssro_AWSopendataNetcdf2ioda.py]: thank you for the comment. Resolved!

**Review by BenjaminRuston — APPROVED** [review]
thanks @hailingz did you test for all the data types in the satellite ID list

**Comment by hailingz** [comment]
> thanks @hailingz did you test for all the data types in the satellite ID list

Do you mean if all existing satellite ID are included in the list?

**Pull request merged**